test-project/Assets/EditmodeTests/GameObjectRepresentation/MonoBehaviourActivationManagerTests.cs

```diff
@@ -0,0 +1,347 @@
+using Generated.Improbable;
+using Generated.Improbable.Gdk.Tests.ComponentsWithNoFields;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.Commands;
+using Improbable.Worker.Core;
+using NUnit.Framework;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.GameObjectRepresentation.EditModeTests.MonoBehaviourActivationManagerTests
+{
+    public abstract class ActivationManagerTestBase<TBehaviour> where TBehaviour : Component
```

Comment:
Only testing like this assumes that handling multiple MonoBehaviours in parallel works correctly ie they are completely independent... A few tests with multiple might be good to make sure that works too

Reply:
Do you mean multiple behaviours on same gameobject? I'll handle that too